server/aesdsocket.c

```diff
@@ -0,0 +1,346 @@
+	    	exit(EXIT_FAILURE);
+	    }
+
+            if (chdir("/") == -1)
+	    {
+            	syslog(LOG_ERR, "chdir");
+            	close(socket_fd);
+		closelog();
+            	exit(EXIT_FAILURE);
+            }
+
+    	    int dev_null_fd = open("/dev/null", O_RDWR);
+            dup2(dev_null_fd, STDIN_FILENO);
+            dup2(dev_null_fd, STDOUT_FILENO);
+            dup2(dev_null_fd, STDERR_FILENO);
+
+            close(STDIN_FILENO);
+            close(STDOUT_FILENO);
+            close(STDERR_FILENO);
+    	}
+    }
+    if(listen(socket_fd,BACKLOG_CONNECTIONS) == -1)
+    {
+	syslog(LOG_ERR, "Error in listen\n");
+        close(socket_fd);
+        closelog();
+
+	return -1;
+    }
+
+    struct sockaddr_storage their_addr;
+    socklen_t addr_size;
+    addr_size = sizeof(their_addr);
+
+    // This variable contains total bytes transferred over all connections
+    int check_tot = 0;
+    int realloc_count = 1;
+
+
+        char *buf;
+
+        buf = (char *)malloc(sizeof(char) * MAXDATASIZE);
+
+
+        char * write_buf;
+        write_buf = (char *)malloc(sizeof(char) * MAXDATASIZE);
+
+	write_file_fd = open(FILE_PATH, O_RDWR | O_APPEND | O_CREAT, S_IRWXU);
+
+        if (write_file_fd == -1)
+        {
+        printf("2\n");
+            syslog(LOG_ERR, "Error in opening file");
+            close(socket_fd);
+            closelog();
+            free(buf);
+            free(write_buf);
+            return -1;
+        }
+
+
+
+
+    while(1)
+    {
+
+	client_fd = accept(socket_fd, (struct sockaddr *)&their_addr, &addr_size);
+
+	if(client_fd == -1)
+	{
+		printf("1\n");
+	    syslog(LOG_ERR, "Error in retrieving client fd\n");
+            close(socket_fd);
+            closelog();
+	    return -1;
+	}
+
+	inet_ntop(AF_INET, get_in_addr((struct sockaddr *)&their_addr), ipstr, sizeof ipstr);
+        syslog(LOG_DEBUG,"Accepted connection from %s", ipstr);
+	int numbytes = 0;
+/*
+    	char *buf;
+
+    	buf = (char *)malloc(sizeof(char) * MAXDATASIZE);
+
+
+        char * write_buf;
+        write_buf = (char *)malloc(sizeof(char) * MAXDATASIZE);
+
+*/
+
+
+/*    	write_file_fd = open(FILE_PATH, O_RDWR | O_APPEND | O_CREAT, S_IRWXU);
+
+    	if (write_file_fd == -1)
+    	{
+	printf("2\n");
+            syslog(LOG_ERR, "Error in opening file");
+            close(socket_fd);
+            closelog();
+	    free(buf);
+	    free(write_buf);
+	    return -1;
+    	}*/
+
+	int buf_location = 0;
+
+	do //while((numbytes = recv(client_fd, buf + buf_location, MAXDATASIZE -1, 0)) > 0)
+	{
+	    numbytes = recv(client_fd, buf + buf_location, MAXDATASIZE, 0);
+	    if(numbytes == -1) {
+printf("3\n");
+        	perror("recv");
+		close(socket_fd);
+		closelog();
+		free(buf);
+		free(write_buf);
+        	return -1;
+    	    }
+	    ++realloc_count;
+	    buf_location += numbytes;
+            check_tot +=numbytes;
+	    //char * temp_buf;
+            buf = (char *)realloc(buf, realloc_count*MAXDATASIZE*(sizeof(char)));
```

Comment:
@Sharath-Jonnala what assumption doet his make about the total size of received packets relative to the size of the heap?